src/workerd/api/node/streams-test.js

```diff
@@ -1940,3 +1945,2627 @@ export const finishedTest = {
+        const seen = seenChunks[i];
+        // There was a chunk.
+        ok(seen);
+        const expected = Buffer.from(expectedChunks[i]);
+        // It was what we expected.
+        ok(seen.equals(expected));
+      }
+      queueMicrotask(() => {
+        // The stream should not have been ended.
+        ok(!seenEnd);
+        w.end();
+      });
+    });
+
+    await closed.promise;
+  }
+};
+
+export const corkEnd = {
+  async test(ctrl, env, ctx) {
+    // Test the buffering behavior of Writable streams.
+    //
+    // The call to cork() triggers storing chunks which are flushed
+    // on calling end() and the stream subsequently ended.
+    //
+    // node version target: 0.12
+
+    const closed = deferredPromise();
+    const expectedChunks = ['please', 'buffer', 'me', 'kindly'];
+    const inputChunks = expectedChunks.slice(0);
+    let seenChunks = [];
+    let seenEnd = false;
+    const w = new Writable();
+    // Let's arrange to store the chunks.
+    w._write = function (chunk, encoding, cb) {
+      // Stream end event is not seen before the last write.
+      ok(!seenEnd);
+      // Default encoding given none was specified.
+      strictEqual(encoding, 'buffer');
+      seenChunks.push(chunk);
+      cb();
+    };
+    // Let's record the stream end event.
+    w.on('finish', () => {
+      seenEnd = true;
+    });
+    function writeChunks(remainingChunks, callback) {
+      const writeChunk = remainingChunks.shift();
+      let writeState;
+      if (writeChunk) {
+        queueMicrotask(() => {
+          writeState = w.write(writeChunk);
+          // We were not told to stop writing.
+          ok(writeState);
+          writeChunks(remainingChunks, callback);
+        });
+      } else {
+        callback();
+      }
+    }
+
+    w.on('close', closed.resolve);
+
+    // Do an initial write.
+    w.write('stuff');
+    // The write was immediate.
+    strictEqual(seenChunks.length, 1);
+    // Reset the seen chunks.
+    seenChunks = [];
+
+    // Trigger stream buffering.
+    w.cork();
+
+    // Write the bufferedChunks.
+    writeChunks(inputChunks, () => {
+      // Should not have seen anything yet.
+      strictEqual(seenChunks.length, 0);
+
+      // Trigger flush and ending the stream.
+      w.end();
+
+      // Stream should not ended in current tick.
+      ok(!seenEnd);
+
+      // Buffered bytes should be seen in current tick.
+      strictEqual(seenChunks.length, 4);
+
+      // Did the chunks match.
+      for (let i = 0, l = expectedChunks.length; i < l; i++) {
+        const seen = seenChunks[i];
+        // There was a chunk.
+        ok(seen);
+        const expected = Buffer.from(expectedChunks[i]);
+        // It was what we expected.
+        ok(seen.equals(expected));
+      }
+      queueMicrotask(() => {
+        // Stream should have ended in next tick.
+        ok(seenEnd);
+      });
+    });
+
+    await closed.promise;
+
```

Comment:
There's one stray blank line here: just pointing out because this is overall immaculately written.